File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    var total: usize = 0;
    for (1..number + 1) |i| {
        total += i;
    }
    return total * total;
}

pub fn sumOfSquares(number: usize) usize {
    var total: usize = 0;
    for (1..number + 1) |i| {
        total += i * i;
    }

    return total;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}

File: typescript/difference-of-squares/difference-of-squares.ts
export class Squares {
  private count: number = 0;
  constructor(count: number) {
    this.count = count;
  }

  get sumOfSquares(): number {
    let result: number = 0;
    for(let i = 1; i <= this.count; i++) {
      result += i * i;
    }

    return result;
  }

  get squareOfSum(): number {
    let sum: number = 0;
    for(let i = 1; i <= this.count; i++) {
      sum += i;
    }

    return sum * sum;
  }

  get difference(): number {
    return this.squareOfSum - this.sumOfSquares
  }
}
